src/ai/core/market_predictor.py

```
@@ -0,0 +1,144 @@
+            # Calculate momentum (0 to 1)
+            momentum = abs(recent_growth - historical_growth)
+            momentum = min(1.0, momentum)
+            
+            # Calculate confidence (0 to 1)
+            confidence = min(1.0, abs(volume_change))
+            
+            return PropertyTrend(
+                property_type=performance_data.get('type', 'unknown'),
+                trend_direction=trend_direction,
+                momentum=momentum,
+                confidence=confidence
```

Ruff (0.8.2)

Trailing comma missing

(COM812)